jdbi/src/main/java/org/killbill/commons/jdbi/guice/DataSourceProvider.java

```diff
@@ -90,22 +94,36 @@ public void setHealthCheckRegistry(final HealthCheckRegistry healthCheckRegistry
         this.healthCheckRegistry = healthCheckRegistry;
     }
 
+    @Inject(optional = false)
```

Comment:
Shouldn't it be `true` (only required for `NONE` pooling type)? We could simply add a NPE check in the corresponding branch in `buildDataSource` below.